### ISSUE TITLE
[Bug]: panic in library/core/src/slice/index.rs:52:5

### ISSUE THREAD

**Issue by M4n5ter** [ClementTsang/bottom#786]
### Checklist

- [X] I've looked through [troubleshooting](https://clementtsang.github.io/bottom/nightly/troubleshooting), [known problems](https://clementtsang.github.io/bottom/nightly/support/official/#known-problems), and  [existing open issues](https://github.com/ClementTsang/bottom/issues?q=is%3Aopen+is%3Aissue) for similar issues.

### Describe the issue

I executed `btm` on my terminal(`Tabby`)  before lunch. when I came back, I found this panic.
```shell
~   
❯ btm
thread '<unnamed>' panicked at 'range start index 2 out of range for slice of length 1', library/core/src/slice/index.rs:52:5
   0: bottom::panic_hook
   1: std::panicking::rust_panic_with_hook
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/std/src/panicking.rs:702:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/std/src/panicking.rs:588:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/std/src/sys_common/backtrace.rs:138:18
   4: rust_begin_unwind
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/std/src/panicking.rs:584:5
   5: core::panicking::panic_fmt
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/panicking.rs:142:14
   6: core::slice::index::slice_start_index_len_fail_rt
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/slice/index.rs:52:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/ops/function.rs:248:5
   8: core::intrinsics::const_eval_select
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/intrinsics.rs:2372:5
   9: core::slice::index::slice_start_index_len_fail
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/slice/index.rs:42:9
  10: <bottom::canvas::Painter as bottom::canvas::widgets::disk_table::DiskTableWidget>::draw_disk_table
  11: bottom::canvas::Painter::draw_widgets_with_constraints
  12: core::iter::traits::iterator::Iterator::for_each
  13: bottom::canvas::Painter::draw_data::{{closure}}
  14: bottom::try_drawing
  15: btm::main
  16: std::sys_common::backtrace::__rust_begin_short_backtrace
  17: main
  18: __libc_start_main
             at /build/glibc-2.28.19/csu/../csu/libc-start.c:308:16
  19: _start
```
```shell
~   
❯ neofetch
             ............                i78@i78-PC 
         .';;;;;.       .,;,.            ---------- 
      .,;;;;;;;.       ';;;;;;;.         OS: Deepin 20.6 x86_64 
    .;::::::::'     .,::;;,''''',.       Host: 82MS Yoga 14sACH 2021 
   ,'.::::::::    .;;'.          ';      Kernel: 5.10.101-amd64-desktop 
  ;'  'cccccc,   ,' :: '..        .:     Uptime: 2 days, 17 mins 
 ,,    :ccccc.  ;: .c, '' :.       ,;    Packages: 2328 (dpkg) 
.l.     cllll' ., .lc  :; .l'       l.   Shell: zsh 5.7.1 
.c       :lllc  ;cl:  .l' .ll.      :'   Resolution: 2880x1800 
.l        'looc. .   ,o:  'oo'      c,   DE: Deepin 
.o.         .:ool::coc'  .ooo'      o.   WM: KWin 
 ::            .....   .;dddo      ;c    Theme: deepin-dark [GTK2/3] 
  l:...            .';lddddo.     ,o     Icons: bloom-dark [GTK2/3] 
   lxxxxxdoolllodxxxxxxxxxc      :l      Terminal: Tabby 
    ,dxxxxxxxxxxxxxxxxxxl.     'o,       CPU: AMD Ryzen 7 5800H with Radeon Graphics (16) @ 3.200GHz 
      ,dkkkkkkkkkkkkko;.    .;o;         GPU: AMD ATI Cezanne 
        .;okkkkkdl;.    .,cl:.           Memory: 10002MiB / 15413MiB 
            .,:cccccccc:,.
```

### What operating system are you using?

Linux

### What architecture are you using?

x86_64 / AMD64

### What terminal(s) are you running `bottom` on?

tabby 1.0.183

### What version of `bottom` are you running?

0.6.8

### How did you install `bottom`?

cargo install bottom --locked

### How can we reproduce this?

just run `btm` and wait for 1h(maybe).

**Comment by ClementTsang** [comment]
Likely a duplicate of #773. It's fixed in one of my branches.

**Comment by ClementTsang** [comment]
Should be closed by #749.
